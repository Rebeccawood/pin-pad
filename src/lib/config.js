const hexStringToRGB = (color) => {
  const result = color.match(/([a-f\d]{2})([a-f\d]{2})([a-f\d]{2})/i);
  return `${parseInt(result[1], 16)}, ${parseInt(result[2], 16)}, ${parseInt(
    result[3],
    16
  )}`;
};

export default {
  resolutions: {
    mobileL: "500px",
  },
  palette: {
    green: "#4b5435",
    greenTransparent(alpha) {
      return `rgba(${hexStringToRGB(this.green)}, ${alpha})`;
    },
    pink: "#d90755",
    pinkTransparent(alpha) {
      return `rgba(${hexStringToRGB(this.pink)}, ${alpha})`;
    },
    white: "#ffffff",
    whiteTransparent(alpha) {
      return `rgba(${hexStringToRGB(this.white)}, ${alpha})`;
    },
    offWhitePrimary: "#e8dee6",
    offWhiteSecondary: "#e8dde5",
    grey: "#7a7d7b",
  },

  borderRadius: {
    desktop: "2rem",
    mobile: "1rem",
  },
  border: {
    green: "0.1rem solid #4b5435",
    white: "0.1rem solid white",
  },
};
