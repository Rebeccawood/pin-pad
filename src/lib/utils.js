const hideNumbers = (digits) => {
  const displayNumberString = digits.join("");
  const masked = displayNumberString.substring(
    0,
    displayNumberString.length - 1
  );
  const lastNum = displayNumberString.substr(displayNumberString.length - 1);
  const hiddenNumbers = masked.replace(/./g, "*");
  return hiddenNumbers.concat(lastNum);
};

const buttonDigits = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9];

const checkPin = (pin, digits) =>
  JSON.stringify(pin) === JSON.stringify(digits);

const showCorrectMessage = (pinChecked) => (pinChecked ? "OK!" : "ERROR");

export { hideNumbers, buttonDigits, checkPin, showCorrectMessage };
