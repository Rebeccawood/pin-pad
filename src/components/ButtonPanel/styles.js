import styled from "styled-components";
import config from "../../lib/config";

const Wrapper = styled.section`
  display: flex;
  justify-content: center;
  margin-top: 8rem;

  @media (max-width: ${config.resolutions.mobileL}) {
    display: grid;
    margin-top: 2rem;
    min-height: 100%;
    width: 100%;
  }
`;

const Panel = styled.section`
  background: ${config.palette.whiteTransparent("0.6")};
  border-radius: ${config.borderRadius.desktop};
  border: 0.1rem solid ${config.palette.green};
  column-gap: 1rem;
  display: grid;
  grid-template-columns: 8rem 8rem 8rem;
  grid-template-rows: auto;
  padding: 5rem;
  row-gap: 1.5rem;

  @media (max-width: ${config.resolutions.mobileL}) {
    border-radius: ${config.borderRadius.mobile};
    grid-template-columns: 5rem 5rem 5rem 5rem;
    padding: 1rem;
    row-gap: 1rem;
    width: 100%;
  }
`;

const Button = styled.div`
  background-color: ${config.palette.green};
  border-radius: ${config.borderRadius.desktop};
  border: ${config.border.white};
  box-shadow: ${config.palette.grey} 1rem 0rem 1rem 0rem;
  color: ${config.palette.white};
  cursor: pointer;
  font-size: 3rem;
  padding: 1.5rem;

  &:hover {
    transform: scale(1.1);
  }

  @media (max-width: ${config.resolutions.mobileL}) {
    border-radius: ${config.borderRadius.mobile};
    font-size: 1.6rem;
    padding: 1rem;
  }
`;

export { Wrapper, Panel, Button };
