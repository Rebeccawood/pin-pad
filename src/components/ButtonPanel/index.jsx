import React from "react";
import { Wrapper, Panel, Button } from "./styles";
import { buttonDigits } from "../../lib/utils";

const ButtonPanel = (props) => {
  const { displayNumber, setDisplayNumber } = props;

  return (
    <Wrapper>
      <Panel>
        {buttonDigits.map((num) => (
          <Button
            key={num}
            onClick={() => setDisplayNumber([...displayNumber, num])}
          >
            {num}
          </Button>
        ))}
        <Button onClick={() => setDisplayNumber([])}>X</Button>
      </Panel>
    </Wrapper>
  );
};

export default ButtonPanel;
