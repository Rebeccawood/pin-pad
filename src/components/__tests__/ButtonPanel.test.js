import React from "react";
import ButtonPanel from "../ButtonPanel/index.jsx";
import "@testing-library/jest-dom/extend-expect";
import Adapter from "enzyme-adapter-react-16";
import { configure } from "enzyme";
import renderer from "react-test-renderer";
import { buttonDigits } from "../../lib/utils";

configure({ adapter: new Adapter() });

describe("ButtonPanel Component", () => {
  it("matches the snapshot", () => {
    const tree = renderer.create(<ButtonPanel />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("checks there are 10 digit buttons", () => {
    expect(buttonDigits.length).toBe(10);
  });
});
