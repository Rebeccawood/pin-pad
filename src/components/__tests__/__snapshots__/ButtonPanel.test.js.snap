// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonPanel Component matches the snapshot 1`] = `
<section
  className="sc-AxjAm deJgmo"
>
  <section
    className="sc-AxirZ hFvKcS"
  >
    <div
      className="sc-AxiKw jfrPL"
      onClick={[Function]}
    >
      0
    </div>
    <div
      className="sc-AxiKw jfrPL"
      onClick={[Function]}
    >
      1
    </div>
    <div
      className="sc-AxiKw jfrPL"
      onClick={[Function]}
    >
      2
    </div>
    <div
      className="sc-AxiKw jfrPL"
      onClick={[Function]}
    >
      3
    </div>
    <div
      className="sc-AxiKw jfrPL"
      onClick={[Function]}
    >
      4
    </div>
    <div
      className="sc-AxiKw jfrPL"
      onClick={[Function]}
    >
      5
    </div>
    <div
      className="sc-AxiKw jfrPL"
      onClick={[Function]}
    >
      6
    </div>
    <div
      className="sc-AxiKw jfrPL"
      onClick={[Function]}
    >
      7
    </div>
    <div
      className="sc-AxiKw jfrPL"
      onClick={[Function]}
    >
      8
    </div>
    <div
      className="sc-AxiKw jfrPL"
      onClick={[Function]}
    >
      9
    </div>
    <div
      className="sc-AxiKw jfrPL"
      onClick={[Function]}
    >
      X
    </div>
  </section>
</section>
`;
